### PR TITLE
[CDAP-8191] Revamps global navbar based on new design

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds ProductDropdown, MetadataDropdown components to be used in Header components
- Revamps Header to be more responsive to different view sizes
- Fixes responsiveness of cards in home page based on design inputs.

JIRA: https://issues.cask.co/browse/CDAP-8191
Bamboo build: http://builds.cask.co/browse/CDAP-DRC5527-6

### CODE REVIEW

File: cdap-ui/app/cdap/components/Header/index.js [hunk 1]
```diff
@@ -15,60 +15,142 @@
...
+            !this.state.toggleNavbar ?
+              <i className="fa fa-bars fa-2x"></i>
+            :
+              <i className="fa fa-times fa-2x"></i>
+          }
+        </div>
+        <div className="brand-section">
+          <img
+            src="/cdap_assets/img/company_logo.png"
+            height="50px"
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/Header/index.js]: style with CSS instead?

File: cdap-ui/app/cdap/components/CustomDropdownMenu/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+
+const CustomDropdownMenu = (props, context) => {
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/CustomDropdownMenu/index.js:20]: why do we need a CustomDropdownMenu? if it's just the ability to change the Tag, reactstrap already has that capability, right?

File: cdap-ui/app/cdap/components/Header/ProductDropdown/index.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            <a
+              target="_blank"
+              href="http://cask.co/community"
+            >
+              {T.translate('features.Navbar.ProductDropdown.supportLabel')}
+            </a>
+          </DropdownItem>
+          <DropdownItem tag="li">
+            <a
+              href="http://docs.cask.co"
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/Header/ProductDropdown/index.js]: shouldn't this go to the specific cdap docs version?

**Comment by elfenheart** [comment]
screen shot 2017-02-06 at 3 04 23 pm

On hover behavior in the angular app portion seems different


Also, How do I add CDAP entity now?

**Comment by tonybach** [comment]
The preferences modal now appears under the namespace dropdown, instead of over it like in the current UI:

screen shot 2017-02-06 at 3 16 42 pm

Not sure if it's because part of my old code is incompatible with the new navbar.

**Comment by bdmogal** [comment]
@tonybach the namespace dropdown should be closed after the user clicks on the preferences fast action, right? So we should never be in this state?

File: cdap-ui/app/cdap/components/CustomDropdownMenu/index.js (hunk 2 above)
Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/CustomDropdownMenu/index.js:20]: Not right now. It doesn't have a `tag` prop in 3.9.4.

**Comment by ajainarayanan** [comment]
@elfenheart The Resource Center `+` button will be added at a level below the top navbar.

**Comment by elfenheart** [comment]
@ajainarayanan if we are adding at a level below the top navbar, then wouldn't we have to add it in every single page? We can add it part of navbar and add position absolute or fixed

**Comment by tonybach** [comment]
@bdmogal Didn't know that was the behavior that we wanted, can create a quick PR for it.

@ajainarayanan Sorry to hijack your PR lol.

**Comment by bdmogal** [comment]
@tonybach @ajainarayanan but that behavior makes sense, right? Let's report a jira?

**Comment by elfenheart** [comment]
@bdmogal I thought keeping it open was intentional, we even added a success notification on the cards when preference is saved, right?

**Comment by bdmogal** [comment]
@elfenheart we have success notification on the cards in the dropdown? I'll try it out

**Comment by elfenheart** [comment]
@bdmogal yes we do. 

@ajainarayanan one thing I noticed, adding a new namespace if we are in pipeline view, it doesn't update the namespace dropdown.

**Comment by bdmogal** [comment]
Ok. yeah I saw it in the morning. Let's keep it, but fix the problem that @tonybach reported above?

**Comment by elfenheart** [comment]
namespace dropdown z-index is still greater than preference modal

**Comment by elfenheart** [comment]
👍  LGTM

File: cdap-ui/app/cdap/styles/variables.scss [hunk 4]
```diff
@@ -20,6 +20,7 @@ $cdap_body_min_width:   700px;
 // new variables for use in themes
 $cdap-white:            rgb(255, 255, 255); // #fff, white
 $cdap-orange:           rgb(255, 102, 0);   // #ff6600
+$cdap-light-orange:     rgb( 255, 167, 38);
```
Comment (elfenheart) [inline cdap-ui/app/cdap/styles/variables.scss]: nit: no space before `255`

**Comment by tonybach** [comment]
Other than the above, code LGTM 👍

**Comment by elfenheart** [comment]
👍  LGTM

**Pull request merged**